File: src/Ccmc.Core/Services/ClaudeReadiness.cs
namespace Ccmc.Core.Services;

/// <summary>Cheap "can Claude start cleanly here?" checks. Filesystem only, no network.</summary>
public static class ClaudeReadiness
{
    /// <summary>True if the user's .claude dir is writable, or absent but the home dir is writable.</summary>
    public static bool IsClaudeDirWritable(string? homeDir = null)
    {
        homeDir ??= Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
        if (!Directory.Exists(homeDir)) return false;
        var claudeDir = Path.Combine(homeDir, ".claude");
        var probeDir = Directory.Exists(claudeDir) ? claudeDir : homeDir;
        try
        {
            var probe = Path.Combine(probeDir, ".devprojects-write-probe");
            File.WriteAllText(probe, "");
            File.Delete(probe);
            return true;
        }
        catch (Exception ex) when (ex is IOException or UnauthorizedAccessException)
        {
            return false;
        }
    }
}

File: electron/src/core/claude/claudeReadiness.ts
/**
 * Pure "can Claude start cleanly here?" decision logic.
 *
 * The actual filesystem probe (writing a temp file to verify write access) is a
 * Phase 3 concern handled by the main-process wrapper.
 *
 * TODO Phase 3: wrap isClaudeDirWritable with a real fs probe:
 *   - resolve homeDir via os.homedir()
 *   - pick probeDir = .claude subdir if it exists, else homeDir
 *   - write+delete a temp file; return true on success, false on IOException/EPERM
 */

/** Inputs derived from the filesystem by the Phase 3 caller. */
export interface ClaudeReadinessFacts {
  /** Whether the home directory itself exists. */
  homeExists: boolean
  /** Whether $HOME/.claude exists (if false, probe goes to homeDir). */
  claudeDirExists: boolean
  /** Whether a write+delete probe succeeded in the probe directory. */
  canWrite: boolean
}

/**
 * Pure decision: the user's .claude dir (or home dir) is writable.
 * Returns false immediately when home is missing.
 */
export function isClaudeDirWritable(facts: ClaudeReadinessFacts): boolean {
  if (!facts.homeExists) return false
  return facts.canWrite
}
